File: apps/api/app/api/ai_dossier_drafts.py
from fastapi import APIRouter, Depends, HTTPException
from pydantic import BaseModel

from app.core.tenant_context import get_current_tenant
from app.ai.repositories.dossier_draft_repository import (
    get_dossier_draft,
    list_dossier_drafts,
    update_dossier_draft_status,
)
from app.ai.services.dossier_draft_service import (
    prepare_dossier_draft_from_message,
)
from app.ai.services.dossier_execution_service import execute_dossier_draft


router = APIRouter()


class PrepareDossierDraftRequest(BaseModel):
    source_message: str
    workflow_id: str | None = None
    manager_id: str | None = None
    manager_name: str | None = None


class UpdateDraftStatusRequest(BaseModel):
    status: str


@router.post("/inbox/conversations/{phone}/ai-dossier-draft")
def create_ai_dossier_draft(
    phone: str,
    body: PrepareDossierDraftRequest,
    tenant=Depends(get_current_tenant),
):
    org_id = tenant["org_id"]

    return prepare_dossier_draft_from_message(
        org_id=org_id,
        client_phone=phone,
        source_message=body.source_message,
        workflow_id=body.workflow_id,
        manager_id=body.manager_id,
        manager_name=body.manager_name,
    )


@router.get("/inbox/conversations/{phone}/ai-dossier-drafts")
def get_ai_dossier_drafts(
    phone: str,
    tenant=Depends(get_current_tenant),
):
    org_id = tenant["org_id"]

    return {
        "status": "ok",
        "drafts": list_dossier_drafts(
            org_id=org_id,
            client_phone=phone,
        ),
    }


@router.patch("/ai-dossier-drafts/{draft_id}/status")
def change_ai_dossier_draft_status(
    draft_id: str,
    body: UpdateDraftStatusRequest,
):
    return {
        "status": "ok",
        "draft": update_dossier_draft_status(
            draft_id=draft_id,
            status=body.status,
        ),
    }


@router.post("/ai-dossier-drafts/{draft_id}/execute")
def execute_draft(draft_id: str):
    draft = get_dossier_draft(draft_id)

    if not draft:
        raise HTTPException(
            status_code=404,
            detail="AI dossier draft not found",
        )

    return execute_dossier_draft(draft)

File: apps/web/dashboard/services/ai-dossier-drafts.ts
import { api } from "@/services/api";
import type { AIDossierDraft } from "@/types/inbox";

export async function prepareAIDossierDraft(
  phone: string,
  data: {
    source_message: string;
    workflow_id?: string | null;
    manager_id?: string | null;
    manager_name?: string | null;
  }
): Promise<AIDossierDraft> {
  const response = await api.post(
    `/inbox/conversations/${encodeURIComponent(phone)}/ai-dossier-draft`,
    data
  );

  return response.data.draft;
}

export async function getAIDossierDrafts(
  phone: string
): Promise<AIDossierDraft[]> {
  const response = await api.get(
    `/inbox/conversations/${encodeURIComponent(phone)}/ai-dossier-drafts`
  );

  return response.data.drafts;
}

export async function updateAIDossierDraftStatus(
  draftId: string,
  status: string
): Promise<AIDossierDraft> {
  const response = await api.patch(
    `/ai-dossier-drafts/${draftId}/status`,
    {
      status,
    }
  );

  return response.data.draft;
}

export async function executeDossierDraft(
  draftId: string
) {
  const response = await api.post(
    `/ai-dossier-drafts/${draftId}/execute`
  );

  return response.data;
}

